### PR TITLE
Fix relative path for external-schema

### PR DESCRIPTION
**Pull request by jzaplet**

### CODE REVIEW

File: src/Propel/Generator/Manager/AbstractManager.php [hunk 1]
```diff
@@ -364,6 +364,12 @@ protected function includeExternalSchemas(\DOMDocument $dom, $srcDir)
 
             $externalSchema->parentNode->removeChild($externalSchema);
 
+            $relativePath = realpath($srcDir. '/'. $include);
+
+            if (!file_exists($include) && file_exists($relativePath)) {
```
Comment (marcj) [inline src/Propel/Generator/Manager/AbstractManager.php]: `!file_exists($include)` should be removed tho

Comment (jzaplet) [reply src/Propel/Generator/Manager/AbstractManager.php]: Can I use is_readable function in this condition?

Comment (jzaplet) [reply src/Propel/Generator/Manager/AbstractManager.php]: Or do you think I should remove the first part of the condition?

Comment (marcj) [reply src/Propel/Generator/Manager/AbstractManager.php]: `if (file_exists($relativePath)) {` should be enough

**Comment by dereuromark** [comment]
Is there any test case for this maybe?

**Comment by jzaplet** [comment]
> Is there any test case for this maybe?

If you mean test case of "relative path for external schema".

There is example https://github.com/wakerscz/cms-sandbox/blob/master/app/schema/schema.xml

**Comment by dereuromark** [comment]
Yes. A unit test here would prove that the change is valid and also protect it from accidental regressions in the future.

**Comment by jzaplet** [comment]
But I think, Propel ORM is no longer maintained.

**Comment by dereuromark** [comment]
You can check the recent history and you will see that is very much maintained again. Mainly by Spryker and friends.

**Comment by dereuromark** [comment]
Would be awesome if you also rebased on top of latest master.
As we have now a very strong test suite running for all kind of versions and DBs.

**Comment by mringler** [comment]
Hi @jzaplet,

found this PR when I was struggling with something around external schemas. I am happy to help you write a test case, if you are still interested, and find that this is a valuable addition to Propel.
Looking at the code, I have some questions, maybe we can sort them out and get this going?

File: src/Propel/Generator/Manager/AbstractManager.php [hunk 2]
```diff
@@ -364,6 +364,12 @@ protected function includeExternalSchemas(\DOMDocument $dom, $srcDir)
 
             $externalSchema->parentNode->removeChild($externalSchema);
 
+            $relativePath = realpath($srcDir. '/'. $include);
```
Comment (dereuromark) [inline src/Propel/Generator/Manager/AbstractManager.php:367]: can we use DS constant (DIR... SEP) instead of / maybe?

**Comment by mringler** [comment]
Added a test and some fixes in #1722

**Pull request closed without merging**